app/api/routes/resource_modification.py

```diff
@@ -46,11 +46,19 @@ def update_resource(id, json, db):
         logger.info(
             f"Updating resource. Old data: {json_module.dumps(resource.serialize)}")
         if json.get('languages'):
+            old_languages = resource.languages[:]
             resource.languages = langs
             index_object['languages'] = resource.serialize['languages']
+            for language in old_languages:
+                if not Resource.query.filter_by(languages=language).all():
```

Comment:
I don't like a SQL SELECT being inside a for loop at all. We should be able to select everything we need just once and work with that, right?

It would require more memory to store all the Resources in an array and then search for the specific ones we need.. not sure if the trade-off of time for more memory is the right answer either... 

I think optimizing this will require more research and thinking it through. And perhaps it's not a huge deal because this route isn't used very often, maybe we can worry about optimizing it later.

If we do decide to kick that can down the road, we should leave a comment about it's possible performance issues so when someone (us?) is trying to hunt down the cause of a performance problem later, it's easier to find.

Reply:
I'd rather get it right now. I can change this to query the database once.